src/functions.jl

```diff
@@ -102,6 +102,28 @@ for (mod, funs) in (
     end
 end
 
+# Two arrays
+for (mod, funs) in (
+    (:Base, (:sum!, :prod!, :maximum!, :minimum!)),
+)
+    for fun in funs
+        @eval begin
+
+            function $mod.$fun(a::NamedDimsArray{L}, b::AbstractArray) where L
+                data = $mod.$fun(parent(a), b)
+                return NamedDimsArray{L}(data)
+            end
+
+            function $mod.$fun(a::NamedDimsArray{La}, b::NamedDimsArray{Lb}) where {La, Lb}
+                newL = unify_names_longest(La, Lb)
```

Comment:
why `unify_names_longest` and not just `unify_names`?

Reply:
Because this works: `sum!(zeros(2), rand(2,10))`. 

However it's still not quite right, as I want only `newL[1:ndims(a)]`.